tardis/io/tests/test_HDFReaderWriter.py

```diff
@@ -0,0 +1,105 @@
+import os
+
+import numpy as np
+import pandas as pd
+import pandas.util.testing as pdt
+import pytest
+from astropy import units as u
+from astropy.tests.helper import assert_quantity_allclose
+from numpy.testing import assert_almost_equal, assert_array_almost_equal
+
+from tardis.io.util import HDFReaderWriter
+
+
+class Mock_HDF(HDFReaderWriter, object):
+    hdf_properties = ['property']
+    quantity_attrs = {}
+
+    def __init__(self, property):
+        self.property = property
+
+    def to_hdf(self, file_path, path=''):
+        buff_path = os.path.join(path, 'dummy')
+        self.to_hdf_util(file_path, buff_path, {name: getattr(self, name) for name
+                                                in self.hdf_properties})
+
+    @classmethod
+    def from_hdf(cls, file_path, path=''):
+        buff_path = os.path.join(path, 'dummy')
+        data = cls.from_hdf_util(file_path, buff_path)
+        return cls(**data)
+
+#Test Cases
+
+#DataFrame
+#None
+#Numpy Arrays
+#Strings
+#Numeric Values
+#Pandas Series Object
+#MultiIndex Object
+#Quantity Objects with - Numeric Values , Numpy Arrays , DataFrame , Pandas Series, None objects
+
+
+simple_objects = [1.5, 'random_string', 4.2e7, None]
+
+
+@pytest.mark.parametrize("attr", simple_objects)
+def test_simple_readwrite(tmpdir, attr):
+    fname = str(tmpdir.mkdir('data').join('dummy.hdf'))
+    actual = Mock_HDF(attr)
+    actual.to_hdf(fname, 'Mock_HDF')
+    expected = Mock_HDF.from_hdf(fname, 'Mock_HDF')
+    assert actual.property == expected.property
+
+
+mock_df = pd.DataFrame({'one': pd.Series([1., 2., 3.], index=['a', 'b', 'c']),
+                        'two': pd.Series([1., 2., 3., 4.], index=['a', 'b', 'c', 'd'])})
+complex_objects = [np.array([4.0e14, 2, 2e14, 27.5]),
+                   pd.Series([1., 2., 3.]), mock_df]
+
+
+@pytest.mark.parametrize("attr", complex_objects)
+def test_complex_obj_readwrite(tmpdir, attr):
+    fname = str(tmpdir.mkdir('data').join('dummy.hdf'))
+    actual = Mock_HDF(attr)
+    actual.to_hdf(fname, 'Mock_HDF')
+    expected = Mock_HDF.from_hdf(fname, 'Mock_HDF')
+    assert_array_almost_equal(actual.property, expected.property)
+
+
+arrays = [['L1', 'L1', 'L2', 'L2', 'L3', 'L3', 'L4', 'L4'],
+           ['one', 'two', 'one', 'two', 'one', 'two', 'one', 'two']]
+tuples = list(zip(*arrays))
+mock_multiIndex = pd.MultiIndex.from_tuples(tuples, names=['first', 'second'])
+
+@pytest.mark.xfail
```

Comment:
I first created a dummy `MultiIndex` object from this pandas [documentation](https://pandas.pydata.org/pandas-docs/stable/advanced.html) .
And as I mentioned in the description of the PR , it is not handled yet.